test/Microsoft.Azure.ServiceBus.UnitTests/ExpectedMessagingExceptionTests.cs

```diff
@@ -41,39 +47,68 @@ public class ExpectedMessagingExceptionTests
-            Assert.NotNull((object)messageSession);
-
-            var message = await messageSession.ReceiveAsync();
-            Assert.True(message.MessageId == messageId);
-            TestUtility.Log($"Received Message: SessionId: {messageSession.SessionId}");
-
-            // Let the Session expire
-            await Task.Delay(TimeSpan.FromMinutes(1));
-
-            // Complete should throw
-            await Assert.ThrowsAsync<SessionLockLostException>(async () => await message.CompleteAsync());
             try
             {
-                await messageSession.CloseAsync();
+                var messageId = "test-message1";
+                var sessionId = Guid.NewGuid().ToString();
+                await queueClient.SendAsync(new BrokeredMessage() { MessageId = messageId, SessionId = sessionId });
+                TestUtility.Log($"Sent Message: {messageId} to Session: {sessionId}");
+
+                MessageSession messageSession = await queueClient.AcceptMessageSessionAsync(sessionId);
+                Assert.NotNull((object)messageSession);
+
+                var message = await messageSession.ReceiveAsync();
+                Assert.True(message.MessageId == messageId);
+                TestUtility.Log($"Received Message: SessionId: {messageSession.SessionId}");
+
+                // Let the Session expire
+                await Task.Delay(TimeSpan.FromMinutes(1));
+
+                // Complete should throw
+                await Assert.ThrowsAsync<SessionLockLostException>(async () => await message.CompleteAsync());
+                try
+                {
+                    await messageSession.CloseAsync();
+                }
+                catch (Exception e)
+                {
+                    TestUtility.Log($"Got Exception on Session Close(): SessionId: {messageSession.SessionId}, Exception: {e.Message}");
+                }
+
+                messageSession = await queueClient.AcceptMessageSessionAsync(sessionId);
+                Assert.NotNull((object)messageSession);
+
+                message = await messageSession.ReceiveAsync();
+                TestUtility.Log($"Received Message: SessionId: {messageSession.SessionId}");
+
+                await message.CompleteAsync();
             }
-            catch (Exception e)
+            finally
             {
-                TestUtility.Log($"Got Exception on Session Close(): SessionId: {messageSession.SessionId}, Exception: {e.Message}");
+                await queueClient.CloseAsync();
             }
+        }
 
-            messageSession = await queueClient.AcceptMessageSessionAsync(sessionId);
-            Assert.NotNull((object)messageSession);
-
-            message = await messageSession.ReceiveAsync();
-            TestUtility.Log($"Received Message: SessionId: {messageSession.SessionId}");
+        [Fact]
+        async Task CompleteOnPeekedMessagesShouldThrowTest()
```

Comment:
Just curious - why the `Test` suffix? This code is not going to be executed in production and it's attributed as a test (`[Fact]`).